plugins/tasks/tasks-core/src/com/intellij/tasks/gitlab/GitlabRepository.java

```diff
@@ -300,4 +310,74 @@ private HttpGet getApiVersionRequest() {
   public void setProjects(@NotNull List<GitlabProject> projects) {
     myProjects = projects;
   }
+
+
+  @Override
+  protected int getFeatures() {
+    return super.getFeatures() | TIME_MANAGEMENT;
+  }
+
+  /**
+   * Adds time spent to a task.
+   *
+   * @param task      The local task we are submitting time for.
+   * @param timeSpent The amount of time spent on the issue in the format 0h0m.
+   * @param comment   The comment to also add to the issue.
+   * @throws Exception
+   */
+  @Override
+  public void updateTimeSpent(@NotNull LocalTask task, @NotNull String timeSpent, @NotNull String comment) throws Exception {
+    final Pattern issueURLPattern = Pattern.compile("https://.*?/(.*?/.*?)/.*"); // Captures project namespace from URL
+    final String issueURL = task.getIssueUrl();
+    if (issueURL == null) {
+      throw new IllegalArgumentException("A GitLab-bound LocalTask should not have a null issue url.");
+    }
+
+    final Matcher issueURLMatcher = issueURLPattern.matcher(issueURL);
+    if (!issueURLMatcher.matches()) {
+      throw new IllegalStateException("Could not find project namespace from issue URL.");
+    }
+    final String projectNamespace = issueURLMatcher.group(1);
+
+    // Use URL-encoded project namespace since we can't find the project id from a LocalTask
+    final URIBuilder timeUpdateBuilder =
+      new URIBuilder(getRestApiUrl("projects", projectNamespace, "issues", task.getNumber(), "add_spent_time"))
+        .addParameter("duration", timeSpent);
+
+    LOG.debug("Sending POST request to " + timeUpdateBuilder.build());
+
+    final HttpPost timeUpdateRequest = new HttpPost(timeUpdateBuilder.build());
```

Comment:
Please build constructed URLs right away instead of passing `URIBuilder` around.